### ISSUE TITLE
Graph with mAP, Precision and Recall equal to 0 - Object Detection

### ISSUE THREAD

**Issue by caleale90** [NVIDIA/DIGITS#1673]
Hi,
I followed the tutorial for object detection with DIGITS without any problem.
Now I want to train a model to detect dresses from my own dataset.
Here I show an image of my dataset.

![img_00000003](https://user-images.githubusercontent.com/29305603/26975261-4b731522-4d1f-11e7-9a3f-ccba4bdfb3c2.jpg)

For each jpg file I have a txt file in the label folder of the following format. 
I wrote a python script to convert the original format to the DIGITS-friendly format.

Here an example:
dress 0.00 0 0.00 41.00 97.00 244.00 295.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00

Then I create a new object detection dataset and everything is ok.
In these 2 pictures, I show the model settings that I use.
![model1](https://user-images.githubusercontent.com/29305603/26975402-ddf247ba-4d1f-11e7-88f4-a871f5738bdf.png)
![model2](https://user-images.githubusercontent.com/29305603/26975403-ddf249d6-4d1f-11e7-8539-7170291392c2.png)

Instead of using a DetectNet from Pretrained model tab, I tried also to copy and paste the prototxt file from https://raw.githubusercontent.com/NVIDIA/caffe/caffe-0.15/examples/kitti/detectnet_network.prototxt and downloaded the GoogLeNet caffemodel from https://github.com/BVLC/caffe/tree/rc3/models/bvlc_googlenet and then started the training process, but the result is the same and can be seen in these pictures.
![training](https://user-images.githubusercontent.com/29305603/26978604-637c0e82-4d2c-11e7-98e2-6d3fe693ef76.png)
![training2](https://user-images.githubusercontent.com/29305603/26978629-7bd5c2f2-4d2c-11e7-83c8-15d90a431817.png)

Obviously at the end of the train process, if I try to test an image, DIGITS can't find any bounding box as it has had problem during the training step.

Can someone help me to discover any error? Thank you

**Comment by ontheway16** [comment]
Hi I guess you pad your image size to 512x512, right?

**Comment by caleale90** [comment]
Yes, I forgot to explicit this detail, but each image is padded to 512x512

**Comment by ontheway16** [comment]
@caleale90  Does your dataset contain single class (ie, dress) and nothing else? Did you check for typo in label names in dataset? It s best to be sure about integrity of dataset. Majority of errors are coming from this point.

**Comment by caleale90** [comment]
Yes, my dataset contains only one class which is "dress". I am sure that I don't have any spelling mistakes in my .txt files because they are generated using a python script which I attach below.

`import os

from shutil import rmtree
from shutil import copyfile
from itertools import izip


def get_second_part():
    return "0.00 0 0.00"


def get_last_part():
    return "0.00 0.00 0.00 0.00 0.00 0.00 0.00"


def partition_dataset_with_file(source_dir, dest_dir, partition_file, bbox_file):
    with open(partition_file, "r") as partition_file, open(bbox_file, "r") as bbox_file:
        for pf, bf in izip(partition_file, bbox_file):
            pf_row = ' '.join(pf.split()).strip().split(" ")
            bf_row = ' '.join(bf.split()).strip().split(" ")
            if not (len(pf_row) == 1) and not (str(pf_row[0]).startswith("image_name")):
                img_path = pf_row[0]
                phase = pf_row[1]
                img_filename = img_path.split("/")[-1]
                class_value = "dress"
                coords = get_coordinates(bf_row)
                manage_folder(dest_dir + "/" + phase + "/images")
                manage_folder(dest_dir + "/" + phase + "/labels")
                copyfile(source_dir + "/" + img_path, dest_dir + "/" + phase + "/images/" + img_filename)
                with open(dest_dir + "/" + phase + "/labels/" + img_filename.replace(".jpg", ".txt"),
                          "a") as label_file:
                    label_file.write("{} {} {} {}\n".format(class_value, get_second_part(), coords, get_last_part()))


def get_coordinates(bf_row):
    bf_row[1] = "{0:.2f}".format(float(bf_row[1]))
    bf_row[2] = "{0:.2f}".format(float(bf_row[2]))
    bf_row[3] = "{0:.2f}".format(float(bf_row[3]))
    bf_row[4] = "{0:.2f}".format(float(bf_row[4]))
    return "{} {} {} {}".format(bf_row[1], bf_row[2], bf_row[3], bf_row[4])


def manage_folder(dir):
    if not os.path.exists(dir):
        os.makedirs(dir)


def main():
    ROOT_DIR = "/home/super/datasets/Fashion-Landmark-Detection-Benchmark"
    dataset_dir = ROOT_DIR + "/DIGITS-dataset"
    partition_file = ROOT_DIR + "/Eval/list_eval_partition.txt"
    bbox_file = ROOT_DIR + "/Anno/list_bbox.txt"

    # TODO REMOVE
    if os.path.exists(dataset_dir):
        rmtree(dataset_dir)
    # END REMOVE

    manage_folder(dataset_dir)

    partition_dataset_with_file(ROOT_DIR, dataset_dir, partition_file, bbox_file)


if __name__ == '__main__':
    main()
`

Apart from other image manipulation, this code writes the .txt files converting from the original format to DIGITS-like format.
Here an example. 
Starting from this file that contains bounding boxes for each image:
123016
image_name  x_1  y_1  x_2  y_2
img/img_00000001.jpg 053 130 289 440
img/img_00000002.jpg 053 095 207 208
img/img_00000003.jpg 041 097 244 295
...

the code creates for each row, a file named img_********.txt in this format:
dress 0.00 0 0.00 53.00 130.00 289.00 440.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00

I read that DIGITS has a special value for UNDEFINED, I put 0 to say it. I don't know if it's this one the error.

**Comment by ontheway16** [comment]
It appears correct to me. Can you please mention about batch size, accumulation  & learning rate? Also wondering which pretained model you are using? Trained with same or similar dataset?

**Comment by caleale90** [comment]
These are my settings.
I use these batch size because I am using 3 GPUs for training.
![model-settings](https://user-images.githubusercontent.com/29305603/27037244-9baa6fa8-4f87-11e7-8cc0-4904b8544b1f.png)

**Comment by ontheway16** [comment]
I dont know how long it takes to finish single epoch with your dataset, but you may try batch 6 or 9 (if they fit into memory). Also, try to lower learning rate to 0.00001. Did you try to run without pretained model?

**Comment by caleale90** [comment]
Yes, I tried and I obtained same result.
Probably I found the solution. 
During the creation of the dataset, in the custom classes field, I put "dontcare, dress" and it gives at least values for mAP, Precision and Recall different from 0.
Now I got error while training...but I will open a new issue :) This one is fixed

**Comment by YoungSharp** [comment]
Hi ,

I had already put "dontcare,car,dog" in Custom classes . But during the training mAP is still 0......
Do you know how to fix this?
